### DISCUSSION TITLE
How can I add the sampler restart to fooocus?

### DISCUSSION THREAD

**Discussion by elyzionz** [lllyasviel/Fooocus#3151] [Q&A] [answered]
Hello community, can you explain to me how to add new samplers to fooocus? for example I want to add the sampler restart

**Comment by mashb1t** [comment, marked as answer]
you have to extend the array in flags.py, then add the sampler to ldm_patched/k_diffusion/sampling.py

**Reply by dezvin** [reply]
After the upgrade, will the flags.py file code be reset to default?

**Reply by mashb1t** [reply]
no, you'll not be able to automatically update anymore and have to resolve the merge conflicts manually.
